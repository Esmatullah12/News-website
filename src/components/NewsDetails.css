.news-heading {
  display: flex;
  background-color: #5c91f4;
  width: 100%;
  padding: 20px;
  color: white;
  text-align: right;
}

.news-heading img {
  width: 50%;
}

.news-heading h2 {
  margin: 0;
}

.news-detail-page {
  position: relative;
  display: flex;
  max-width: 1200px;
  margin: 0 auto;
}

.new-info {
  font-size: 1.2rem;
  font-weight: 600;
  color: white;
  padding: 10px 20px;
}

.details-date-img {
  max-width: 90%;
  box-shadow: rgba(0, 0, 0, 0.2) 0 20px 30px;
}

.details-page-date {
  margin-top: 10px;
  color: white;
}

.details-page-title {
  font-size: 1.6rem;
  margin-bottom: 30px;
}

.details-page-desc {
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 100px;
  padding: 20px;
  width: 100%;
  color: white;
}

.details-page-desc:nth-child(even) {
  background-color: #4b75c6;
}

.details-page-desc a {
  color: white;
  font-weight: 600;
}

@media screen {
  .news-detail-page {
    flex-direction: column;
  }
}
