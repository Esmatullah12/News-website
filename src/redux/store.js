import { configureStore } from '@reduxjs/toolkit';
import newsSlice from './news/newsSlice';

const store = configureStore({
  reducer: {
    news: newsSlice,
  },
});

export default store;
