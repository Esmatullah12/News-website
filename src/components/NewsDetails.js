import React from 'react';
import { useSelector } from 'react-redux';
import { Link, useParams } from 'react-router-dom';
import './NewsDetails.css';

const NewsDetails = () => {
  const params = useParams();
  const { newsId } = params;
  const targetNews = useSelector((state) => state.news.news
    .find((nNews) => nNews.newsId === parseInt(newsId, 10)));
  if (!targetNews) {
    return <div>Product Not Found</div>;
  }
  const format = {
    year: 'numeric', month: 'long', day: 'numeric', weekday: 'long',
  };
  const formattedDate = new Date(targetNews.newsDate).toLocaleDateString(undefined, format);

  return (
    <>
      <Link className="back-btn" to="/">Back To Home</Link>
      <div className="news-detail-page">
        <img className="details-date-img" src={targetNews.newsImg} alt="News Article" />
        <div className="details-news-info">
          <h2 className="details-page-title">{targetNews.newsTitle}</h2>
          <p className="details-page-date">{formattedDate}</p>
          <p className="details-page-desc">{targetNews.newsDesc}</p>
          <p>
            Author:
            {' '}
            {targetNews.newsAuthor}
          </p>
          <p>
            Original source click
            {' '}
            <a href={targetNews.newsSource}>here</a>
          </p>
        </div>
      </div>
    </>
  );
};

export default NewsDetails;
