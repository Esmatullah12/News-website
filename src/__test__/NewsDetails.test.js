import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import NewsDetails from '../components/NewsDetails';
import store from '../redux/store';

describe('Test NewsDetails component', () => {
  it('it should render NewsDetails component correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <NewsDetails />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
