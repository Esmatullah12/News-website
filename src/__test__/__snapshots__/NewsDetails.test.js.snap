// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NewsDetails component it should render NewsDetails component correctly 1`] = `
<div>
  <div>
    Product Not Found
  </div>
</div>
`;
